src/imputors/fill.jl

```diff
@@ -5,37 +5,38 @@ Fills in the missing data with a specific value.
 
 # Fields
 * `value::Any`: A scalar missing value or a function that returns the a scalar if
-    passed the data with missing data removed (e.g, `mean`)
+  passed the data with missing data removed (e.g, `mean`)
+* `context::AbstractContext`: A context which keeps track of missing data
+  summary information
 """
 struct Fill{T} <: Imputor
     value::T
+    context::AbstractContext
 end
 
-"""
-    Fill() -> Fill
-
-By default `Fill()` will use the mean of the existing values as the fill value.
-"""
-Fill() = Fill(mean)
+"""Fill(; value=mean, context=Context()) -> Fill"""
+Fill(; value=mean, context=Context()) = Fill(value, context)
 
 """
-    impute!(imp::Fill, ctx::Context, data::AbstractVector)
+    impute!(imp::Fill, data::AbstractVector)
 
 Computes the fill value if `imp.value` is a `Function` (i.e., `imp.value(drop(copy(data)))`)
 and replaces all missing values in the `data` with that value.
 """
-function impute!(imp::Fill, ctx::Context, data::AbstractVector)
-    fill_val = if isa(imp.value, Function)
-        imp.value(Iterators.drop(copy(data)))
-    else
-        imp.value
-    end
+function impute!(imp::Fill, data::AbstractVector)
```

Comment:
Am i right in saying this one only works on vectors?
Or is there some overload I am missign that will call it to run on the columns of a table etc?

Reply:
By default, the abstract `Imputor` will call `impute!` for each column in a table or matrix, so in many univariate cases you only need to implement the 1 overloaded method.